engine/engine.go

```diff
@@ -354,10 +355,10 @@ func (e *Engine) RegisterMultihashLister(mhl provider.MultihashLister) {
 // registered.
 //
 // See: Engine.RegisterMultihashLister, Engine.Publish.
-func (e *Engine) NotifyPut(ctx context.Context, contextID []byte, md metadata.Metadata) (cid.Cid, error) {
+func (e *Engine) NotifyPut(ctx context.Context, provider string, addrs []string, contextID []byte, md metadata.Metadata) (cid.Cid, error) {
```

Comment:
Take `peer.AddrInfo` instead of `string` as `provider` and `[]string` as `addrs`?